ui/src/app/pages/access/maybe-module.tsx

```diff
@@ -0,0 +1,80 @@
+// Renders a module when it's enabled via feature flags.  Returns null if not.
+import * as React from 'react';
+import { useState } from 'react';
+
+import { AccessModule, Profile } from 'generated/fetch';
+
+import { WithSpinnerOverlayProps } from 'app/components/with-spinner-overlay';
+import { switchCase } from 'app/utils';
+import {
+  getAccessModuleStatusByName,
+  isEligibleModule,
+  redirectToControlledTraining,
+  redirectToNiH,
+  redirectToRas,
+  redirectToRegisteredTraining,
+} from 'app/utils/access-utils';
+import { AnalyticsTracker } from 'app/utils/analytics';
+import { useNavigation } from 'app/utils/navigation';
+
+import { Module } from './module';
+import { TwoFactorAuthModal } from './two-factor-auth-modal';
+
+interface ModuleProps {
+  profile: Profile;
+  moduleName: AccessModule;
+  active: boolean;
+  clickable: boolean;
+  spinnerProps: WithSpinnerOverlayProps;
+}
+
+export const MaybeModule = ({
```

Comment:
this does not include the `showRefresh` logic ... was that moved to Module?

Reply:
Yes, the showRefresh logic is in Module now. I believe that it can be moved in here, because unless a module has a refreshAction, it will not show the Refresh component.